vllm/engine/multiprocessing/client.py

```diff
@@ -506,15 +508,17 @@ async def beam_search(
                             cum_logprob=current_beam.cum_logprob +
                             logprob_obj.logprob)
 
-                        if token_id == tokenizer.eos_token_id and \
-                            not ignore_eos:
+                        if result.outputs[0].finish_reason == "stop":
                             completed.append(new_beam)
                         else:
                             new_beams.append(new_beam)
 
             sorted_beams = sorted(new_beams, key=sort_beams_key, reverse=True)
             all_beams = sorted_beams[:beam_width]
 
+            if not all_beams:
+                break
```

Comment:
Was this a bug?

Better to move it a couple of lines up and use `if not new_beams`?

Reply:
I'll be honest I got confused by the similar check in `llm.py` and decided to do something similar here. Not sure if it is needed, and we can do as you say, or we can leave it outside altogether.

Reply:
OK thanks, I guess we should figure out which is correct based on the prior/expected behavior...